doc/developer-notes.md

```diff
@@ -250,6 +250,7 @@ $ valgrind --suppressions=contrib/valgrind.supp src/test/test_bitcoin
 $ valgrind --suppressions=contrib/valgrind.supp --leak-check=full \
       --show-leak-kinds=all src/test/test_bitcoin --log_level=test_suite
 $ valgrind -v --leak-check=full src/bitcoind -printtoconsole
+$ ./test/functional/test_runner.py --valgrind
```

Comment:
Thoughts:
- This should wait for #17639
- This is example is visible when running `.test/functional/test_runner.py --help`, which is not the case for the existing examples
- There should be a dedicated valgrind doc to link to here, and/or sections in test/README.md and test/functional/README.md about running with valgrind
- In general, this section could use more documentation, updating, or a link to more details
- This example requires valgrind 3.14 or later, which is not the case for the existing examples, and that should be documented